cassandra/cluster.py

```diff
@@ -3925,20 +3926,15 @@ def _set_result(self, host, connection, pool, response):
                         self._metrics.on_unavailable()
                     retry = retry_policy.on_unavailable(
                         self.query, retry_num=self._query_retries, **response.info)
-                elif isinstance(response, OverloadedErrorMessage):
+                elif isinstance(response, (OverloadedErrorMessage,
+                                           IsBootstrappingErrorMessage,
+                                           TruncateError, ServerError)):
+                    log.warning("Host %s error: %s. retrying against a different host", host, response.summary)
```

Comment:
NIT: Log message states that it's retrying and it is now up to the retry policy to decide.